partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -0,0 +1,663 @@
+package org.partiql.lang.planner.transforms
+
+import com.amazon.ionelement.api.field
+import com.amazon.ionelement.api.ionString
+import com.amazon.ionelement.api.ionStructOf
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.ast.UNKNOWN_SOURCE_LOCATION
+import org.partiql.lang.ast.passes.SemanticProblemDetails
+import org.partiql.lang.errors.Problem
+import org.partiql.lang.errors.ProblemCollector
+import org.partiql.lang.planner.PlanningProblemDetails
+import org.partiql.lang.planner.transforms.PartiQLSchemaInferencerTests.TestCase.ErrorTestCase
+import org.partiql.lang.planner.transforms.PartiQLSchemaInferencerTests.TestCase.SuccessTestCase
+import org.partiql.plugins.mockdb.LocalPlugin
+import org.partiql.spi.sources.ColumnMetadata
+import org.partiql.spi.sources.ValueDescriptor
+import org.partiql.types.AnyOfType
+import org.partiql.types.AnyType
+import org.partiql.types.BoolType
+import org.partiql.types.StaticType
+import org.partiql.types.StaticType.Companion.unionOf
+import org.partiql.types.StructType
+import java.net.URL
+import java.time.Instant
+import java.util.stream.Stream
+import kotlin.test.assertTrue
+
+class PartiQLSchemaInferencerTests {
```

Comment:
Does the code can handle `JOIN`s correctly today? If so, can we add the related tests (could be another PR). Otherwise, let's add an issue to track the `JOIN` support.

Reply:
Decided to add JOINs.